packages/react-storage/src/components/StorageBrowser/composables/DataTable/DataTable.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+
+import { EmptyMessage } from '../../components/EmptyMessage';
+import { LoadingIndicator } from '../../components/LoadingIndicator';
+import { Table } from '../../components/Table';
+import { SortHeader } from './headers/SortHeader';
+import { TextHeader } from './headers/TextHeader';
+import { ButtonDataCell } from './dataCells/ButtonDataCell';
+import { DateDataCell } from './dataCells/DateDataCell';
+import { NumberDataCell } from './dataCells/NumberDataCell';
+import { CheckboxDataCell } from './dataCells/CheckboxDataCell';
+import { TextDataCell } from './dataCells/TextDataCell';
+import { DataTableDataCell, DataTableHeader } from './types';
+import { WithKey } from '../../components/types';
+import { CheckboxHeader } from './headers/CheckboxHeader';
+
+export interface DataTableRow {
+  content: WithKey<DataTableDataCell>[];
+}
+
+export interface DataTableProps {
+  headers: WithKey<DataTableHeader>[];
+  rows: WithKey<DataTableRow>[];
+  isLoading?: boolean;
```

Comment:
Non-blocking: Not sure if the table should know abt loading state -ideally it should just be provided an empty view that would render a loading indicator

Reply:
That makes sense. I wanted to keep the empty message here so we needed some indicator that it's loading - not just actually having no rows. But we can just decouple that from the table as well. I will do that in a followup